rts/Lua/LuaHandle.cpp

```diff
@@ -1443,6 +1443,60 @@ void CLuaHandle::UnitHarvestStorageFull(const CUnit* unit)
 	UnitCallIn(cmdStr, unit);
 }
 
+/*** Called when a unit starts it's self destruct command.
+ *
+ * @function Callins:UnitSelfDestructStarted
+ * @param unitID integer
+ * @param unitDefID integer
+ * @param unitTeam integer
+ */
+void CLuaHandle::UnitSelfDestructStarted(const CUnit* unit) 
+{
+	static const LuaHashString cmdStr(__func__);
+	UnitCallIn(cmdStr, unit);
+}
+
+/*** Called when a unit cancel's it's self destruct command.
+ *
+ * @function Callins:UnitSelfDestructCancelled
+ * @param unitID integer
+ * @param unitDefID integer
+ * @param unitTeam integer
+ */
+void CLuaHandle::UnitSelfDestructCancelled(const CUnit* unit) 
+{
+	static const LuaHashString cmdStr(__func__);
+	UnitCallIn(cmdStr, unit);
+}
+
+/*** Called every second while a unit has a self destruct command active.
+ *
+ * @function Callins:UnitSelfDestructProgress
+ * @param unitID integer
+ * @param unitDefID integer
+ * @param unitTeam integer
+ * @param remainingSeconds float
+ */
+void CLuaHandle::UnitSelfDestructProgress(const CUnit* unit, float remainingSeconds) 
+{
+	RECOIL_DETAILED_TRACY_ZONE;
+	LUA_CALL_IN_CHECK(L);
+	luaL_checkstack(L, 8, __func__);
+
+	const LuaUtils::ScopedDebugTraceBack traceBack(L);
+
+	static const LuaHashString cmdStr(__func__);
+	if (!cmdStr.GetGlobalFunc(L))
+		return;
+
+	lua_pushnumber(L, unit->id);// unitID
+	lua_pushnumber(L, unit->unitDef->id); // unitDefID
+	lua_pushnumber(L, unit->team); // unitTeam
+	lua_pushnumber(L, remainingSeconds); // updatePeriodSeconds
+
+	RunCallInTraceback(L, cmdStr, 2, 0, traceBack.GetErrFuncIdx(), false);
```

Comment:
There's 4 input args
```suggestion
	RunCallInTraceback(L, cmdStr, 4, 0, traceBack.GetErrFuncIdx(), false);
```